### PR TITLE
[dev] AB#92442 [React Library] Refactor UI Library for Dropdown Menus and Dropdown Buttons

### PR DESCRIPTION
**Pull request by mkemper86**
This fixes a bug in allowing the `DropdownMenu` inside the `DropdownButton` to automatically close when the page is clicked outside of the button. It also adds a couple of props to allow overriding the horizontal positioning of the `DropdownMenu` to accommodate some updated designs that have right-aligned menus.

### CODE REVIEW

File: src/inputs/dropdownMenu/dropdownMenu.jsx [hunk 1]
```diff
@@ -230,7 +245,7 @@ function DropdownMenu(props) {
         onDropdownMenuReposition();
 
         return () => {
-            if (isOpen) {
+            if (!isOpen) {
```
Comment (groberts314) [inline src/inputs/dropdownMenu/dropdownMenu.jsx:248]: Is this change intentional?  Can I understand why?

**Review by groberts314 — COMMENTED** [review]
Mostly looks okay.  Just one question.

File: src/inputs/dropdownMenu/dropdownMenu.jsx (hunk 1 above)
Comment (mkemper86) [reply src/inputs/dropdownMenu/dropdownMenu.jsx:248]: Yes it's intentional. We want the listener events to be active when the menu is _open_ because we want to be able to close the menu by clicking on the document outside the dropdown. If the events are removed when the menu is open, then the `onClickOutside` will never get run, right?

Comment (groberts314) [reply src/inputs/dropdownMenu/dropdownMenu.jsx:248]: Yeah, you're making sense.  Like we want to remove the events when it is closed.  I wonder was this just a long-running bug?

**Review by groberts314 — APPROVED** [review]

File: src/inputs/dropdownMenu/dropdownMenu.jsx (hunk 1 above)
Comment (mkemper86) [reply src/inputs/dropdownMenu/dropdownMenu.jsx:248]: I think so...the deprecated `DropdownMenu` works correctly with similar code, so I think it was just an oversight.

**Review by War777 — APPROVED** [review]
